### PR TITLE
intacct c1 dependance field

### PR DESCRIPTION
**Pull request by DhaaraniCIT**
## Summary by CodeRabbit

- **New Features**
  - Enhanced configuration import fields with improved conditional logic and styling adjustments for a more intuitive user experience.
  - Added support for handling dependent field warnings and custom field saving in Intacct import settings.

- **Bug Fixes**
  - Resolved issues with conditional rendering in configuration import fields to ensure correct display based on input states.

- **Refactor**
  - Updated constructor dependencies to include `HelperService` for improved utility functions and cleaner code within components.

- **Style**
  - Introduced new CSS classes in configuration import fields for better visual feedback on input states.
  - Added a custom property for checkbox label text in the theme stylesheet to maintain consistent styling.

### CODE REVIEW

File: src/app/integrations/intacct/intacct-shared/intacct-c1-import-settings/intacct-c1-import-settings.component.ts [hunk 1]
```diff
@@ -60,12 +61,16 @@ export class IntacctC1ImportSettingsComponent implements OnInit {
...
   showDialog: boolean;
 
-  customField: ExpenseField;
+  customField: any;
 
   customFieldControl: AbstractControl;
 
   private sessionStartTime = new Date();
 
+  costCodeFieldOption = [{ attribute_type: 'custom_field', display_name: 'Create a Custom Field', source_placeholder: null, is_dependent: true }];
```
Comment (ashwin1111) [inline src/app/integrations/intacct/intacct-shared/intacct-c1-import-settings/intacct-c1-import-settings.component.ts:70]: setence case - ignoring if taken care

File: src/app/integrations/intacct/intacct-shared/intacct-c1-import-settings/intacct-c1-import-settings.component.ts [hunk 2]
```diff
@@ -60,12 +61,16 @@ export class IntacctC1ImportSettingsComponent implements OnInit {
...
+
+  costCategoryOption = [{ attribute_type: 'custom_field', display_name: 'Create a Custom Field', source_placeholder: null, is_dependent: true }];
```
Comment (ashwin1111) [inline src/app/integrations/intacct/intacct-shared/intacct-c1-import-settings/intacct-c1-import-settings.component.ts:72]: same as above

**Comment by DhaaraniCIT** [comment]
https://github.com/fylein/fyle-integrations-app/assets/55541808/bad21ffd-0dd5-4035-b4dc-0aab4b4ad397

Screenshot 2024-06-13 at 11 27 24 AM

**Comment by DhaaraniCIT** [comment]
https://github.com/fylein/fyle-integrations-app/assets/55541808/ac571ca2-20f1-4c57-9182-46a67676fef5

Screenshot 2024-06-13 at 2 29 20 PM

**Comment by ashwin1111** [comment]
will review once latest comments on slack is resolved

**Comment by DhaaraniCIT** [comment]
https://github.com/fylein/fyle-integrations-app/assets/55541808/b3a49f70-8fba-426c-ab6f-0c548cc6585c

**Pull request merged**